src/python/pants/util/frozendict.py

```diff
@@ -15,13 +27,32 @@ class FrozenDict(Mapping[K, V]):
     are not safe to use.
     """
 
-    def __init__(self, item: Optional[Union[Mapping[K, V], Iterable[Tuple[K, V]]]] = None) -> None:
-        """Creates a `FrozenDict` from a mapping object or a sequence of tuples representing
-        entries.
+    @overload
+    def __init__(self, __items: Iterable[Tuple[K, V]], **kwargs: V) -> None:
+        ...
+
+    @overload
+    def __init__(self, __other: Mapping[K, V], **kwargs: V) -> None:
+        ...
+
+    @overload
+    def __init__(self, **kwargs: V) -> None:
+        ...
+
+    def __init__(
+        self, *item: Optional[Union[Mapping[K, V], Iterable[Tuple[K, V]]]], **kwargs: V
```

Comment:
Maybe rename to `*args` to comply with Python conventions. I was never wedded to `item`, just didn't know what to call it.

Not a big deal to keep, though.

Reply:
I think you misunderstand the convention. As I understand it *args and **kwargs are for pass through. Here we use *items ourselves and items is a good name.